### PR TITLE
Make header compilable in C99 and C11 pedantic mode

### PR DESCRIPTION
**Pull request by chfast**
Resolves https://github.com/ethereum/evmc/issues/19.

### CODE REVIEW

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -245,411 +281,469 @@ enum evmc_status_code
...
+ *  @param code_size   The length of the bytecode.
+ *  @return            All execution results.
+ */
 typedef struct evmc_result (*evmc_execute_fn)(struct evmc_instance* instance,
-                                             struct evmc_context* context,
-                                             enum evmc_revision rev,
-                                             const struct evmc_message* msg,
-                                             uint8_t const* code,
-                                             size_t code_size);
+                                              struct evmc_context* context,
```
Comment (chfast) [inline include/evmc/evmc.h]: I kept this param packing. How do you like it? I think it's useful for documentation purposes.

**Comment by axic** [comment]
It's really hard to see the changes. Is it only C++-style to C-style comments?

**Comment by chfast** [comment]
Only comments, some `,` removed. The file is mostly comments.

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -1,141 +1,177 @@
...
-/// 2. Host -- an entity controlling the EVM. The Host requests code execution
-///            and responses to EVM queries by callback functions.
-///
-/// @defgroup EVMC EVMC
-/// @{
+/**
+ * EVMC: Ethereum Client-VM Connector API
+ *
+ * @copyright
+ * Copyright 2018 Pawel Bylica.
```
Comment (axic) [inline include/evmc/evmc.h]: Btw, if there's a copyright here then I guess all the authors should be listed.

Probably way better is to have an `AUTHORS` file to list everyone, while the header can list the major contributors.

Comment (chfast) [reply include/evmc/evmc.h]: I share the same opinion, but getting a clarification would be nice. See https://github.com/ethereum/evmc/pull/1.

**Pull request merged**